### DISCUSSION TITLE
DependencyEndpoint macro can't derive default when using typed throws

### DISCUSSION THREAD

**Discussion by Mika5652** [pointfreeco/swift-dependencies#259] [General]
Hi, I wanted to try typed throws in my client using the `DependencyClient` macro. I got an error, as you can see in the following image. I am curious if a macro can even handle this conversion from any Error to a specific type. I am afraid not, but I am still not experienced with macro creation. Thanks for the response!

<img width="1106" alt="Snímek obrazovky 2024-08-17 v 17 55 37" src="https://github.com/user-attachments/assets/a6359c9f-5396-4b18-b1d2-f586744ff685">

It seems like the default value in this case works, so it's not a problem in the end.
<img width="578" alt="Snímek obrazovky 2024-08-17 v 18 06 10" src="https://github.com/user-attachments/assets/ce85c883-01b6-4f9f-934b-27bda2c97b92">

**Comment by mbrandonw** [comment]
Hi @Mika5652, thanks for bringing this to our attention. We don't have a fix for this right now, though hopefully will by the time Swift 6 is officially released.

In the meantime you can provide the closure manually to work around the compilation error:

```swift
var saveData: @Sendable () throws(SomeClientError) -> Void = { 
  reportIssue(…)
  throw SomeClientError()
}
```

**Reply by Mika5652** [reply]
Sounds good, thanks for info!

**Comment by dioKaratzas** [comment]
@mbrandonw Have there been any updates on this? Is there any issue to track this?

**Reply by mbrandonw** [reply]
Hi @dioKaratzas, we don't really consider this to be an issue with the library. The `@DependencyClient` macro makes some simplifying assumptions to do its job, and typed throws just don't play nicely with those assumptions. If you want to have typed throws you will need to implement the closure yourself.

The one thing we could do though is diagnose this problem and produce an error from the macro. If anyone wants to help us with that we'd love to have a PR.

**Reply by dioKaratzas** [reply]
Got it, thanks for clarifying @mbrandonw 🙏. That makes sense. I can see how the macro’s assumptions don’t really align with typed throws. A diagnostic from the macro would definitely be helpful to surface the limitation earlier, so I’ll look into whether I can contribute a PR for that.
